model-optimizer/mo/ops/strided_slice.py

```diff
@@ -97,55 +54,86 @@ def convert(attr):
+            strides = node.in_port(3).data.get_value()
+            if strides is None:
+                raise Error('StridedSlice operation supports only constant strides input')
+        else:
+            strides = np.ones_like(begin)
+
+        shape = node.in_port(0).data.get_shape()
+        value = node.in_port(0).data.get_value()
+        input_rank = len(shape)
+        assert len(begin) == len(end) == len(strides), 'begin, end, and strides must be of the same length'
+
+        if shape is None or any([x < 0 for x in shape]):
+            return
+
+        # masks alligment all masks should initial slice_rank (mask_alligment != mask_extending)
+        extend_mask = lambda mask, val=0: np.append(mask, [val] * (len(begin) - len(mask))).astype(int)
+        new_axis_mask = extend_mask(node.new_axis_mask)
+        shrink_axis_mask = extend_mask(node.shrink_axis_mask)
+        begin_mask = extend_mask(node.begin_mask, 1)  # differs from the case when we unroll ellipsis
+        end_mask = extend_mask(node.end_mask, 1)  # when we unroll ellipsis zeros should be inserted
+        # no need to extend ellipsis
+
+        # unroll ellipsis
+        if np.any(node.ellipsis_mask):
+            i = np.nonzero(node.ellipsis_mask)
+            assert len(i[0]) == 1, 'only one nonzero value in ellipsis_mask is allowed'
+            ellipsis_start = i[0][0]
+            # since we don't expect values in begin, end values and take all range of values along ellipsis_start axis
+            begin_mask[ellipsis_start] = 0
+            end_mask[ellipsis_start] = 0
+
+            num = input_rank - len(begin) + np.count_nonzero(node.new_axis_mask[ellipsis_start:])
+            unroll_ellipsis = lambda mask, val=0: np.insert(mask, ellipsis_start + 1, [val] * num).astype(int)
+
+            new_axis_mask = unroll_ellipsis(new_axis_mask)
+            shrink_axis_mask = unroll_ellipsis(shrink_axis_mask)
+            begin_mask, end_mask = unroll_ellipsis(begin_mask), unroll_ellipsis(end_mask)
+            begin, end, strides = unroll_ellipsis(begin), unroll_ellipsis(end), unroll_ellipsis(strides, 1)
+
+        # from now slices are without ellipsis
+        slice_rank = len(begin)
+        slices = [[]] * slice_rank
+        in_idx = 0  # index along input tensor shapes, note that input_rank not necessary is equal to slice_rank
+        for i in range(slice_rank):
+            if new_axis_mask[i]:
+                slices[i] = np.newaxis
+            elif shrink_axis_mask[i]:
+                slices[i] = int(begin[i])
+                if slices[i] < 0:
+                    slices[i] += int(shape[in_idx])
+            else:
+                start, stop = begin[i], end[i]
+                if not begin_mask[i]:  # if begin, and end are not specified take whole range
+                    start = 0 if strides[i] > 0 else -1
+                if not end_mask[i]:
+                    stop = shape[in_idx] if strides[i] > 0 else -shape[in_idx] - 1
```

Comment:
Breaks reshape-ability?

Reply:
This line is not present anymore. But it didn't break reshape-ability. But information to take the whole range stored in `begin/end masks` is still present in op attributes and will be delivered to ngraph. But for shape calculation inside MO we need concrete values, this is what this code did.

In the current version I specify magic `slice(None, None)` instead of shape upper/lower bound but still when we take the whole range `get_shape_from_slice` replaces resulting shape with concrete values. So summing it up, I think here reshape-ability is not broken. @jane-intel what do you think about this?